### PR TITLE
Employee ServiceTest/facade, beanMapperTest

### PR DESCRIPTION
**Pull request by JozeFe**
EmployeeService comments
EmployeeService password authentication test
Employee Facade interface and implementation
EmployeeFindByEmail function
BeanMapperTest, Vehicle Year field, can't be mapped with dozer because java.time.Year doesn't have public contructor, only private, now this field is excluded in mapping

### CODE REVIEW

File: fleet-management-service/src/test/java/cz/fi/muni/pa165/service/BeanMappingServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * @author Jozef Krcho
+ */
+@ContextConfiguration(classes = ServiceConfiguration.class)
+public class BeanMappingServiceImplTest extends AbstractTestNGSpringContextTests {
+
+    @Inject
+    private BeanMappingService beanMappingService;
+
+    @Test
+    public void testEmployeeMapping() {
```
Comment (trebi) [inline fleet-management-service/src/test/java/cz/fi/muni/pa165/service/BeanMappingServiceImplTest.java]: I think writing such a test is completely worthless. Sure, they have it in the [PA165 example project](https://github.com/fi-muni/PA165/blob/master/eshop-service/src/test/java/cz/fi/muni/pa165/service/BeanMappingServiceTest.java), but certainly they are not testing every entity. It resembled me a test of the anemic entity (entity with only getters and setters, but no business logic - like we have in our project), I'ts just lot of writing, but business logic is really tested. I would not bother by writing such tests, delete whole file and rely that implementation of Dozer is correct.

File: fleet-management-service/src/main/java/cz/fi/muni/pa165/service/JpaService.java [hunk 2]
```diff
@@ -93,4 +93,20 @@ public void remove(T entity) {
             throw new FleetManagementDAException("Entity delete error.", ex);
         }
     }
+
+    /**
+     * Removed saved entity by id.
```
Comment (trebi) [inline fleet-management-service/src/main/java/cz/fi/muni/pa165/service/JpaService.java]: Removes/Remove

File: fleet-management-service/src/main/java/cz/fi/muni/pa165/service/EmployeeServiceImpl.java [hunk 3]
```diff
@@ -43,7 +51,12 @@ public boolean authenticate(Employee employee, String password) {
...
     }
 
-    //https://crackstation.net/hashing-security.htm#javasourcecode
+    /**
+     * source: https://crackstation.net/hashing-security.htm#javasourcecode
+     * Create encrypted password in format iterations:salt:hash
+     * @param password unencrypted password
+     * @return encrypted password
+     */
     private static String createHash(String password) {
```
Comment (trebi) [inline fleet-management-service/src/main/java/cz/fi/muni/pa165/service/EmployeeServiceImpl.java]: Nice implementation of the password hashing/validating :+1:

File: fleet-management-service/src/test/java/cz/fi/muni/pa165/service/BeanMappingServiceImplTest.java (hunk 1 above)
Comment (JozeFe) [inline fleet-management-service/src/test/java/cz/fi/muni/pa165/service/BeanMappingServiceImplTest.java]: They don't have test like this, I think this is allright ... If I don't do this thes. We wouldn't find that Vehicle java.time.Year can't be mapped ... we need different field type.

**Comment by JozeFe** [comment]
The fact that BeanMappingServiceImplTest find out, that java.time.Year can't be mapped with dozer, is great reason to have this test.

File: fleet-management-service/src/test/java/cz/fi/muni/pa165/service/BeanMappingServiceImplTest.java (hunk 1 above)
Comment (trebi) [inline fleet-management-service/src/test/java/cz/fi/muni/pa165/service/BeanMappingServiceImplTest.java]: (we discussed it on the slack)

**Pull request merged**